fireworks/core/launchpad.py

```diff
@@ -313,6 +315,47 @@ def add_wf(self, wf, reassign_all=True):
         self.m_logger.info('Added a workflow. id_map: {}'.format(old_new))
         return old_new
 
+    def add_wfs(self, wfs):
+        """
+        Adds a list of workflows to the fireworks database
+        using insert_many for both the fws and wfs, is
+        more efficient than adding them one at a time.
+
+        Args:
+            wfs ([Workflow]): list of workflows or fireworks
+
+        Returns:
+            None
+
+        """
+        # Make all fireworks workflows
+        wfs = [Workflow.from_firework(wf) if isinstance(wf, Firework)
+               else wf for wf in wfs]
+
+        # Initialize new firework counter, starting from the next fw id
+        total_num_fws = sum([len(wf.fws) for wf in wfs])
+        new_fw_counter = self.fw_id_assigner.find_one_and_update(
+            {}, {'$inc': {'next_fw_id': total_num_fws}})['next_fw_id']
+        for wf in tqdm(wfs):
+            # Reassign fw_ids and increment the counter
+            old_new = dict(zip(
+                wf.id_fw.keys(),
+                range(new_fw_counter, new_fw_counter + len(wf.fws))))
+            wf._reassign_ids(old_new)
+            new_fw_counter += len(wf.fws)
+
+            # Set root fws to READY
+            for fw_id in wf.root_fw_ids:
+                wf.id_fw[fw_id].state = 'READY'
+                wf.fw_states[fw_id] = 'READY'
+
+        # Insert all fws and wfs, do workflows first so fws don't
+        # get checked out prematurely
+        self.workflows.insert_many([wf.to_db_dict() for wf in wfs])
+        all_fws = chain.from_iterable([wf.fws for wf in wfs])
```

Comment:
If you want to chase performance when this scales to 1,000s, you'll want to use generator expressions not list comprehensions.  So `chain.from_iterable(wf.fws for wf in wfs)` without the square brackets